lib/element.ts

```diff
@@ -0,0 +1,1055 @@
+   * var value = element.all(by.css('.items li')).reduce(function(acc, elem) {
+   *   return elem.getText().then(function(text) {
+   *     return acc + text + ' ';
+   *   });
+   * }, '');
+   *
+   * expect(value).toEqual('First Second Third ');
+   *
+   * @param {function(number, ElementFinder, number, Array.<ElementFinder>)}
+   *     reduceFn Reduce function that reduces every element into a single value.
+   * @param {*} initialValue Initial value of the accumulator.
+   * @return {!webdriver.promise.Promise} A promise that resolves to the final
+   *     value of the accumulator.
+   */
+  reduce(reduceFn: Function, initialValue: any): webdriver.Promise {
+    var valuePromise = webdriver.promise.fulfilled(initialValue);
+    return this.asElementFinders_().then(function(arr: ElementFinder[]) {
+      return arr.reduce(
+          function(
+              valuePromise: any, elementFinder: ElementFinder, index: number) {
+            return valuePromise.then(function(value: any) {
+              return reduceFn(value, elementFinder, index, arr);
+            });
+          },
+          valuePromise);
+    });
+  }
+
+  /**
+   * Evaluates the input as if it were on the scope of the current underlying
+   * elements.
+   *
+   * @view
+   * <span class="foo">{{variableInScope}}</span>
+   *
+   * @example
+   * var value = element.all(by.css('.foo')).evaluate('variableInScope');
+   *
+   * @param {string} expression
+   *
+   * @return {ElementArrayFinder} which resolves to the
+   *     evaluated expression for each underlying element.
+   *     The result will be resolved as in
+   *     {@link webdriver.WebDriver.executeScript}. In summary - primitives will
+   *     be resolved as is, functions will be converted to string, and elements
+   *     will be returned as a WebElement.
+   */
+  evaluate(expression: string) {
+    var evaluationFn = function(webElem: webdriver.WebElement) {
+      return webElem.getDriver().executeScript(
+          clientSideScripts.evaluate, webElem, expression);
+    };
+    return this.applyAction_(evaluationFn);
+  }
+
+  /**
+   * Determine if animation is allowed on the current underlying elements.
+   * @param {string} value
+   *
+   * @example
+   * // Turns off ng-animate animations for all elements in the <body>
+   * element(by.css('body')).allowAnimations(false);
+   *
+   * @return {ElementArrayFinder} which resolves to whether animation is allowed.
+   */
+  allowAnimations(value: boolean): ElementArrayFinder {
+    var allowAnimationsTestFn = function(webElem: webdriver.WebElement) {
+      return webElem.getDriver().executeScript(
+          clientSideScripts.allowAnimations, webElem, value);
+    };
+    return this.applyAction_(allowAnimationsTestFn);
+  }
+}
+
+export class ElementFinder {
+  parentElementArrayFinder: ElementArrayFinder;
+  elementArrayFinder_: ElementArrayFinder;
+  then: Function = null;
+
+  /**
+   * The ElementFinder simply represents a single element of an
```

Comment:
Move the jsdoc for the constructor above the `export class ElementFinder`. Dgeni will not pick up this comment when it is transpiled...